sk0-back/src/ormconfig.ts

```diff
@@ -1,42 +1,44 @@
-import { DataSource, DataSourceOptions } from "typeorm";
+import { DataSourceOptions } from "typeorm";
 
-const { NODE_ENV, DB_NAME } = process.env;
+export function dataSourceOptionFactory(
+  dbName: string,
+  nodeEnv: string
+): DataSourceOptions {
+  let dataSourceOptions: DataSourceOptions;
 
-let dataSourceOptions: DataSourceOptions;
-
-switch (NODE_ENV) {
-  case "test":
-    dataSourceOptions = {
-      type: "sqlite",
-      database: ":memory:",
-      entities: ["src/**/*.entity.ts"],
-      synchronize: true
-    };
-    break;
-  case "development":
-    dataSourceOptions = {
-      type: "sqlite",
-      database: "dev.sqlite",
-      entities: ["dist/**/*.entity.js"],
-      synchronize: true
-    };
-    break;
-  case "production":
-    dataSourceOptions = {
-      type: "sqlite",
-      database: "db.sqlite",
-      entities: ["dist/**/*.entity.js"],
-      synchronize: false,
-      migrations: [__dirname + "/migrations/*.{ts,js}"]
-    };
-    break;
-  default:
-    dataSourceOptions = {
-      type: "sqlite",
-      database: DB_NAME || "sample.sqlite",
-      entities: ["dist/**/*.entity.js"],
-      synchronize: true
-    };
+  switch (nodeEnv) {
+    case "test":
+      dataSourceOptions = {
+        type: "sqlite",
+        database: ":memory:",
+        entities: ["src/**/*.entity.ts"],
+        synchronize: true
+      };
+      break;
+    case "development":
+      dataSourceOptions = {
+        type: "sqlite",
+        database: dbName || "dev.sqlite",
+        entities: ["dist/**/*.entity.js"],
+        synchronize: true
+      };
+      break;
+    case "production":
+      dataSourceOptions = {
+        type: "postgres",
+        database: "db.sqlite",
+        entities: ["dist/**/*.entity.js"],
+        synchronize: false,
+        migrations: [__dirname + "/migrations/*.{ts,js}"]
+      };
+      break;
+    default:
+      dataSourceOptions = {
+        type: "sqlite",
+        database: dbName || "sample.sqlite",
+        entities: ["dist/**/*.entity.js"],
+        synchronize: true
+      };
+  }
+  return dataSourceOptions;
```

Comment:
I am not the biggest fan of this one and I am probably opinionated. But I consider using long switch-cases hard to understand. Running though the cases we have to remember what happened before getting at the end and returning.

Should we rather have early returns?

Or to avoid the the accumulated cases adding a map rather?

Reply:
Changed to map.